src/solvers/refinement/string_constraint_generator_comparison.cpp

```diff
@@ -191,7 +191,7 @@ string_constraint_generatort::add_axioms_for_hash_code(
   array_poolt &pool)
 {
   PRECONDITION(f.arguments().size() == 1);
-  string_constraintst constraints;
+  string_constraintst hash_constraints;
```

Comment:
:pick: I'd probably rename the member variable to `global_constraints` or `all_constraints`

Reply:
Please forgive me for not intending to act on this one: from reading the code I got the impression that the two cases touched here affect methods that should likely be deprecated or removed. @romainbrenguier Any insights maybe?

Reply:
No problem - all of these comments are optional, I'm not overly familiar with the code being modified here anyway.

Reply:
Yes hash code and intern should be implemented via models rather than internally in the solver (they are very language dependent), so this should disappear.